keps/sig-network/20181024-service-topology.md

```diff
@@ -125,12 +129,25 @@ In our example above, we will firstly try to find the backends in the same host.
 
 If we configure topologyKeys as `["kubernetes.io/hostname", "*"]`, we just do the effort to find the backends in the same host and will not fail the connection if no matched backends found.
 
-### Intersection with Local External Traffic Policy
+##### Intersection with Local External Traffic Policy
 
 Topology Aware Services (# of topologyKeys > 0) and Services using `externalTrafficPolicy=Local` will be mutually exclusive. This will be enforced by API validation where a Service with `externalTrafficPolicy=Local` cannot specify any topology keys and vice versa.
 
 Before Service Topology goes GA, kube-proxy should be able to provide feature parity with `externalTrafficPolicy=Local` when the chosen topology is `kubernetes.io/hostname`. The correctness and feasibility of the `kubernetes.io/hostname` topology as a sufficient replacement for `externalTrafficPolicy=Local` will be evaluated during the alpha phase.
 
+#### Option 2: service.spec.internalTrafficPolicy
+
+Another option is to add a new field `internalTrafficPolicy` to Service that directly addresses the "route to node local endpoints" use-case.
+In comparison to option 1 (adding `topologyKeys`), this field adds less complexity to the API but doesn't allow for users to define arbitrary preferred topologies for their services.
+
+Some policies we would define during the alpha stage of this API would be:
+- Cluster (default): route to all cluster-wide endpoints.
```

Comment:
Could we add another option here that was `PreferZone` and potentially also default? I think that would work well with the idea of a more automatic approach to topology aware routing.

Reply:
I wasn't sure if we want the default to be the "automatic topology" approach, or if we want to keep the default simple (random cluster-wide) and add a new policy for the automatic approach later.

Reply:
To me it makes more sense to keep the default as-is and only change it when we have more confidence in the automatic approach later.

I'm also not sure that `PreferZone` can ever be the default since not all clusters have a cloud provider enabled to apply the zone topology label.

Reply:
Yep, not sure. This is probably sufficient for now. Whenever I get around to writing KEP updates for an automatic approach I can add something like `PreferZone` here. I think the default we choose here will have to depend on how safe any kind of automatic approach is.